tfx/examples/cifar10/meta_data_writer.py

```diff
@@ -0,0 +1,208 @@
+# limitations under the License.
+# ==============================================================================
+"""Writes metadata and label file to the image classifier models."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+from absl import app
+from absl import flags
+import tensorflow as tf
+
+import flatbuffers
+# pylint: disable=g-direct-tensorflow-import
+from tflite_support import metadata as _metadata
+from tflite_support import metadata_schema_py_generated as _metadata_fb
+# pylint: enable=g-direct-tensorflow-import
+
+FLAGS = flags.FLAGS
+
+
+def define_flags():
+  flags.DEFINE_string("model_file", None,
+                      "Path and file name to the TFLite model file.")
+  flags.DEFINE_string("label_file", None, "Path to the label file.")
+  flags.DEFINE_string("export_directory", None,
+                      "Path to save the TFLite model files with metadata.")
+  flags.mark_flag_as_required("model_file")
+  flags.mark_flag_as_required("label_file")
+  flags.mark_flag_as_required("export_directory")
+
+
+class ModelSpecificInfo(object):
+  """Holds information that is specificly tied to an image classifier."""
+
+  def __init__(self, name, version, image_width, image_height, image_min,
+               image_max, mean, std, num_classes):
+    self.name = name
+    self.version = version
+    self.image_width = image_width
+    self.image_height = image_height
+    self.image_min = image_min
+    self.image_max = image_max
+    self.mean = mean
+    self.std = std
+    self.num_classes = num_classes
+
+
+_MODEL_INFO = {
+    "cifar10.tflite":
+        ModelSpecificInfo(
+            name="MobileNetV1 image classifier on cifar 10",
+            version="v1",
+            image_width=224,
+            image_height=224,
+            image_min=0,
+            image_max=255,
+            mean=[127.5],
+            std=[127.5],
+            num_classes=10),
+}
+
+
+class MetadataPopulatorForImageClassifier(object):
+  """Populates the metadata for an image classifier."""
+
+  def __init__(self, model_file, model_info, label_file_path):
+    self.model_file = model_file
+    self.model_info = model_info
+    self.label_file_path = label_file_path
+    self.metadata_buf = None
+
+  def populate(self):
+    """Creates metadata and then populates it for an image classifier."""
+    self._create_metadata()
+    self._populate_metadata()
+
+  def _create_metadata(self):
+    """Creates the metadata for an image classifier."""
+
+    # Creates model info.
+    model_meta = _metadata_fb.ModelMetadataT()
+    model_meta.name = self.model_info.name
+    model_meta.description = ("Identify the most prominent object in the "
+                              "image from a set of %d categories." %
+                              self.model_info.num_classes)
+    model_meta.version = self.model_info.version
+    model_meta.author = "TensorFlow"
```

Comment:
what happens when someone not from the tensorflow generates the metadata?

Reply:
This is just the model's description, i.e. you can define it whatever you want and it will not have any side effects.

Reply:
Sure, but maybe its best to have this be configurable vs hard-coded :)

Reply:
sure sounds good